### ISSUE TITLE
weird behavior with complexity checks and inline declarations

### ISSUE THREAD

**Issue by TDA** [eslint/eslint#3634]
Hello,
So, when I define an inline rule, inside a function like so:

```
/*eslint complexity:7*/
```

I am still able to override this by setting this in the eslintrc file:

```
complexity: [0]
```

However, it will not override if I set it to this in the eslintrc file:

```
complexity: 0
```

I think this is weird, either we should not be able to override at all, or we should be. Using a different syntax should not allow overrides IMO.
This is on grunt-eslint@14.0.0 and eslint@0.22.1

cc'ing @pdehaan

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by pdehaan** [comment]
According to @TDA, he was still able to reproduce this in **eslint@1.3.1**.

**Comment by IanVS** [comment]
The correct format to configure the complexity should be: `/*eslint complexity: [2, 7]*/` Where the 2 sets the rule to error, and 7 is the max complexity. 

That said, I'll play around with this a little to see if I can reproduce the behavior you describe.

**Comment by IanVS** [comment]
Wait a second, I just re-read this.  Inline configuration should always override settings in the `.eslintrc`.

When you talk about overrides, can you give examples of ESLint output for each case, and what you are expecting?

**Comment by TDA** [comment]
@IanVS Glad to hear that that is the expected behavior.  We have/had the issue on this: https://github.com/mozilla/fxa-auth-db-mysql/pull/96
Removing the inline declaration solved our problem, but I think this is a valid bug.
It threw a warning when the `complexity: 7` was inline, and `complexity: 0` is in eslintrc.
But it doesn't throw a warning if `complexity: [0]` is in eslintrc.

**Comment by IanVS** [comment]
Sorry, I don't see what the bug would be, since `complexity: 7` is not a valid configuration.  When I try it, I (correctly) get a thrown error from the config validator:

```
▶ node_modules/.bin/eslint example.js                           
/home/ian/Local_code/eslint/eslint/lib/config-validator.js:93
        throw new Error(message.join(""));
        ^

Error: example.js line 1:
	Configuration for rule "complexity" is invalid:
	Severity should be one of the following: 0 = off, 1 = warning, 2 = error (you passed "3").

    at Object.validateRuleOptions (/home/ian/Local_code/eslint/eslint/lib/config-validator.js:93:15)
    at /home/ian/Local_code/eslint/eslint/lib/eslint.js:315:39
    at Array.forEach (native)
    at /home/ian/Local_code/eslint/eslint/lib/eslint.js:313:44
    at Array.forEach (native)
    at modifyConfigsFromComments (/home/ian/Local_code/eslint/eslint/lib/eslint.js:280:18)
    at EventEmitter.module.exports.api.verify (/home/ian/Local_code/eslint/eslint/lib/eslint.js:712:22)
    at processText (/home/ian/Local_code/eslint/eslint/lib/cli-engine.js:199:27)
    at processFile (/home/ian/Local_code/eslint/eslint/lib/cli-engine.js:224:12)
    at /home/ian/Local_code/eslint/eslint/lib/cli-engine.js:391:26
```

According to https://github.com/mozilla/fxa-auth-db-mysql/blob/master/package.json#L44, It looks like you are using a very old version of `grunt-eslint`, and I would recommend updating to the latest and trying again with a valid configuration.

**Comment by ilyavolodin** [comment]
@TDA Can you try to change your inline declaration to:
```js
/*eslint complexity:[2,7]*/
```
And see if you still have the same issues? `complexity:7` is invalid syntax, first argument in the array should always be severity flag.

**Comment by nzakas** [comment]
Also, if you upgrade ESLint, we added configuration validation in later versions.

**Comment by TDA** [comment]
@ilyavolodin That does solve it, and we do have complexity set to [2, 6] in most of our repos, this particular thing is weird because eslint still threw a warning when `complexity: 7` was declared, much like how it would react to `complexity: [1, 7]`
And I still don't understand why setting `complexity: 0` does not override that, but `complexity: [0]` does. I think its no longer relevant to the later versions, and manually updating the dependency for grunt-eslint fixes that. Thanks!
@nzakas I figured that :D thanks for your response.

**Comment by ilyavolodin** [comment]
@TDA Glad that it helped. I'm not really sure about why `complexity: 0` didn't work. It should. You are correct, it should behave exactly like `complexity: [0]`. But I think manual upgrade solved your issue, so I'm going to close this. Feel free to reopen if you find other instances of this problem.
